paasta_tools/setup_marathon_job.py

```diff
@@ -547,33 +547,47 @@ def setup_service(service, instance, client, marathon_config,
 
 
 def main():
-    """Attempt to set up the marathon service instance given.
-    Exits 1 if the deployment failed.
+    """Attempt to set up a list of marathon service instances given.
+    Exits 1 if any service.instance deployment failed.
     This is done in the following order:
 
     - Load the marathon configuration
     - Connect to marathon
-    - Load the service instance's configuration
-    - Create the complete marathon job configuration
-    - Deploy/bounce the service
-    - Emit an event about the deployment to sensu"""
+    - Do the following for each service.instance:
+        - Load the service instance's configuration
+        - Create the complete marathon job configuration
+        - Deploy/bounce the service
+        - Emit an event about the deployment to sensu"""
+
     args = parse_args()
     soa_dir = args.soa_dir
     if args.verbose:
         logging.basicConfig(level=logging.DEBUG)
     else:
         logging.basicConfig(level=logging.WARNING)
 
-    try:
-        service, instance, _, __ = decompose_job_id(args.service_instance)
-    except InvalidJobNameError:
-        log.error("Invalid service instance specified. Format is service%sinstance." % SPACER)
-        sys.exit(1)
-
     marathon_config = get_main_marathon_config()
     client = marathon_tools.get_marathon_client(marathon_config.get_url(), marathon_config.get_username(),
                                                 marathon_config.get_password())
 
+    num_failed_deployments = 0
+    for service_instance in args.service_instance_list:
+        try:
+            service, instance, _, __ = decompose_job_id(service_instance)
+        except InvalidJobNameError:
+            log.error("Invalid service instance specified. Format is service%sinstance." % SPACER)
+            num_failed_deployments = num_failed_deployments + 1
+        else:
+            if deploy_marathon_service(service, instance, client, soa_dir, marathon_config):
+                num_failed_deployments = num_failed_deployments + 1
+
+    print("%d out of %d service.instances are successfully deployed" %
```

Comment:
I would say log.debug this. This will be running very often and will clutter the logs with many of these messages.

For the most part this command should be "no news is good news" in the normal mode. (non verbose)